src/routes.py

```diff
@@ -14,10 +14,38 @@ def home():
 
 @flask_app.route('/callback', methods=['POST'])
 def callback():
-	data = request.json
-	token = data.get('access_token')
-	logging.info(f'Received token: {token}')
-	return jsonify({"status": "success", "message": "Token received"})
+    try:
+        data = request.json
+        token = data.get('access_token')
+
+        if not token:
+            return jsonify({"status": "error", "message": "No token provided"}), 400
+
+        token_id = store_token(token)
+        logging.info(f'JWT received and stored token with ID {token_id}')
+
+        return jsonify({"status": "success", "message": "Token received and stored", "id": token_id})
+
+    except Exception as e:
+        logging.error(f"An error occurred: {e}")
+        return jsonify({"status": "error", "message": "An error occurred processing your request"}), 500
+
+def store_token(token, filename='tokens.txt'):
```

Comment:
This should be a Redis write, with the option to dump to file. The default should be in memory only with Redis.